src/engine/srv_internal.h

```diff
@@ -272,11 +272,7 @@ sched_create_thread(struct dss_xstream *dx, void (*func)(void *), void *arg,
 	struct sched_info	*info = &dx->dx_sched_info;
 	int			 rc;
 #ifdef ULT_MMAP_STACK
-	bool			 tls_set = dss_tls_get() ? true : false;
-	struct dss_xstream	*cur_dx = NULL;
-
-	if (tls_set)
-		cur_dx = dss_current_xstream();
+	struct dss_xstream	*cur_dx = dss_current_xstream();
```

Comment:
What's the difference after this change?

Reply:
No difference, i am going to do same thing in the dss_free_stack_cb(),  instead of checking it in the caller, i decided to check it inside dss_current_xstream().